spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/DistriOptimizer.scala

```diff
@@ -95,8 +95,8 @@ object DistriOptimizer {
     endWhen: Trigger,
     metrics: Metrics,
     models: RDD[Cache[T]],
-    optimMethod: OptimMethod[T],
-    parameters: AllReduceParameter[T],
+    optimMethods: Map[String, OptimMethod[T]],
+    parameters: Map[String, (Int, Int, AllReduceParameter[T])],
```

Comment:
should we add start offset as attributes of AllReduceParameter? It will make code easier to read

Reply:
Looks like the map can be replaced by Seq